htlcswitch/switch.go

```diff
@@ -398,17 +411,25 @@ func (s *Switch) handlePacketForward(packet *htlcPacket) error {
 			source.ShortChanID(),
 			destination.ShortChanID(),
 			htlc.PaymentHash,
+			packet.obfuscator,
 		)); err != nil {
-			reason := []byte{byte(lnwire.UnknownError)}
+			failure := lnwire.FailPermanentChannelFailure{}
```

Comment:
Not sure this should send a permanent failure, as the error may have been transient.